cmd/thin_entrypoint/main.go

```diff
@@ -224,8 +225,16 @@ func (o *Options) createKubeConfig(prevCAHash, prevSATokenHash []byte) ([]byte,
 	if err != nil {
 		return nil, nil, fmt.Errorf("template parse error: %v", err)
 	}
+	var kubeConfigHost string
+	if ip := net.ParseIP(kubeHost); ip != nil && ip.To4() == nil {
+		// use brackets only for IPv6 address
+		kubeConfigHost = fmt.Sprintf("%s://[%s]:%s", kubeProtocol, kubeHost, kubePort)
+	} else {
+		kubeConfigHost = fmt.Sprintf("%s://%s:%s", kubeProtocol, kubeHost, kubePort)
+	}
```

Comment:
no, just use `net.JoinHostPort(kubeHost, kubePort)`. It handles all this for you.